erpnext/public/js/utils/barcode_scanner.js

```diff
@@ -10,6 +10,11 @@ erpnext.utils.BarcodeScanner = class BarcodeScanner {
 		this.serial_no_field = opts.serial_no_field || "serial_no";
 		this.batch_no_field = opts.batch_no_field || "batch_no";
 		this.qty_field = opts.qty_field || "qty";
+		this.max_qty_field = opts.max_qty_field || null;
+		this.allow_new_row = opts.allow_new_row;
+		if (this.allow_new_row === undefined || this.allow_new_row === null) {
+			this.allow_new_row = true;
+		}
```

Comment:
Using negation of this is simpler. 

`this.dont_allow_new_row` which by default is correct `undefined`.

Reply:
Resolved by : https://github.com/frappe/erpnext/pull/30832/commits/7ae89dedd59815cc0dea63a3ecf67b2355ea4425